### PR TITLE
Debugging Documentation for binding to DynamicProperty

### PR DESCRIPTION
**Pull request by iv-mexx**
This is something that comes up every now an then, e.g. #2775, #2719 or #2425 so I tried to clear this up in the DebuggingTechniques documentation.

### CODE REVIEW

File: Documentation/DebuggingTechniques.md [hunk 1]
```diff
@@ -37,3 +37,31 @@ let disposable = sideEffectProducer.startWithCompleted {
...
+
+This will often result in a compiler error: 
+
+> error: binary operator '<~' cannot be applied to operands of type 'DynamicProperty' and 'SignalProducer<String, NoError>'
+DynamicProperty(object: label, keyPath: "text") <~ property.producer
+
+The reason is a limitation in the swift type checker - A `DynamicProperty` always has a type of `AnyObject?`, but the `<~` operator requires the values of both sides to have the same type, so the right side value would have to be `AnyObject?` as well, but usually a more concrete type is used (in this example `String`).
+
+Usually, the fix is as easy as adding a `.map{$0}`.
+
```
Comment (RuiAAPeres) [inline Documentation/DebuggingTechniques.md]: space between `$0` and brackets is missing.

Comment (iv-mexx) [inline Documentation/DebuggingTechniques.md]: @RuiAAPeres fixed it 👍

**Comment by iv-mexx** [comment]
Oh yeah I forgot about #2838 which might make this here obsolet :sweat_smile:

**Comment by RuiAAPeres** [comment]
True, but it could still be useful for people that are using older versions.

**Comment by RuiAAPeres** [comment]
@iv-mexx  We could also update the documentation and mention #2838 (once it has been merged). Sounds good?

**Comment by iv-mexx** [comment]
@RuiAAPeres Yeah, sounds good! 👍

**Pull request merged**